### PR TITLE
feat(*): add breadcrumb

### PR DESCRIPTION
**Pull request by qiuxiaolian**
**What this PR does / why we need it**:

- bread crunmb nav
screen shot 2018-12-24 at 3 01 59 pm

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @cd1989 @caosm 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: web/src/layout/Breadcrumb.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * define the route rules
+ * list page => /project
+ * project detail  => /project/:projectId
+ * update project => /project/:projectId/update
+ * add project => /project/add
+ */
+const BreadcrumbComponent = ({ location }) => {
+  const pathSnippets = location.pathname.split('/').filter(i => i);
```
Comment (cd1989) [inline web/src/layout/Breadcrumb.js]: What's the purpose of `.filter(i => i)`

File: web/src/layout/Breadcrumb.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { Breadcrumb } from 'antd';
+import PropTypes from 'prop-types';
+
+const mainModules = ['overview', 'project', 'template', 'resource', 'workflow'];
+const operation = ['update', 'add'];
```
Comment (cd1989) [inline web/src/layout/Breadcrumb.js]: operations

File: web/src/layout/Breadcrumb.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+ * project detail  => /project/:projectId
+ * update project => /project/:projectId/update
+ * add project => /project/add
+ */
+const BreadcrumbComponent = ({ location }) => {
+  const pathSnippets = location.pathname.split('/').filter(i => i);
+  const extraBreadcrumbItems = pathSnippets.map((path, index) => {
+    const url = `/${pathSnippets.slice(0, index + 1).join('/')}`;
+    let text = path;
+    if (_.includes(mainModules, `${path}`)) {
```
Comment (cd1989) [inline web/src/layout/Breadcrumb.js]: `includes(mainModules, path)`

File: web/src/layout/Breadcrumb.js (hunk 1 above)
Comment (qiuxiaolian) [reply web/src/layout/Breadcrumb.js]: Filter out empty values

File: web/src/layout/Breadcrumb.js (hunk 3 above)
Comment (qiuxiaolian) [reply web/src/layout/Breadcrumb.js]: i use the `lodash` JavaScript utility library

Comment (cd1989) [reply web/src/layout/Breadcrumb.js]: You mistaken it, i mean `_.includes(mainModules, path)`, use `path` directly.

Comment (qiuxiaolian) [reply web/src/layout/Breadcrumb.js]: ok

File: web/src/layout/Breadcrumb.js (hunk 1 above)
Comment (cd1989) [reply web/src/layout/Breadcrumb.js]: Sounds good

**Pull request merged**